src/app/clusters/webrtc-transport-provider-server/webrtc-transport-provider-server.cpp

```diff
@@ -0,0 +1,501 @@
+{
+    // Extract command fields from the request.
+    uint16_t sessionId = req.webRTCSessionID;
+    auto sdpSpan       = req.sdp;
+
+    NodeId peerNodeId           = GetNodeIdFromCtx(ctx.mCommandHandler);
+    FabricIndex peerFabricIndex = ctx.mCommandHandler.GetAccessingFabricIndex();
+
+    WebRTCSessionTypeStruct * existingSession = FindSession(sessionId);
+
+    if (!existingSession)
+    {
+        ChipLogError(Zcl, "HandleProvideAnswer: No existing session with ID=%u.", sessionId);
+
+        // If WebRTCSessionID is not null and does not match a value in current sessions.
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Protocols::InteractionModel::Status::ConstraintError);
+        return;
+    }
+
+    // Also check that the existing session belongs to the same PeerNodeID / Fabric
+    // If it doesn’t match, respond with CONSTRAINT_ERROR
+    if (peerNodeId != existingSession->peerNodeID || peerFabricIndex != existingSession->peerFabricIndex)
+    {
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Protocols::InteractionModel::Status::ConstraintError);
+        return;
+    }
+
+    std::string sdpAnswer(sdpSpan.data(), sdpSpan.size());
+    ctx.mCommandHandler.AddStatus(
+        ctx.mRequestPath, Protocols::InteractionModel::ClusterStatusCode(mDelegate->HandleProvideAnswer(sessionId, sdpAnswer)));
+}
+
+void WebRTCTransportProviderServer::HandleProvideICECandidates(HandlerContext & ctx,
+                                                               const Commands::ProvideICECandidates::DecodableType & req)
+{
+    // Extract command fields from the request.
+    uint16_t sessionId = req.webRTCSessionID;
+
+    NodeId peerNodeId           = GetNodeIdFromCtx(ctx.mCommandHandler);
+    FabricIndex peerFabricIndex = ctx.mCommandHandler.GetAccessingFabricIndex();
+
+    std::vector<std::string> candidates;
+    auto iter = req.ICECandidates.begin();
+    while (iter.Next())
+    {
+        // Get current CharSpan
+        const chip::CharSpan & candidateSpan = iter.GetValue();
+
+        // Convert to std::string
+        candidates.emplace_back(candidateSpan.data(), candidateSpan.size());
+    }
+
+    WebRTCSessionTypeStruct * existingSession = FindSession(sessionId);
+
+    if (!existingSession)
+    {
+        ChipLogError(Zcl, "HandleProvideICECandidates: No existing session with ID=%u.", sessionId);
+
+        // If WebRTCSessionID is not null and does not match a value in current sessions.
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Protocols::InteractionModel::Status::ConstraintError);
+        return;
+    }
+
+    // Also check that the existing session belongs to the same PeerNodeID / Fabric
+    // If it doesn’t match, respond with CONSTRAINT_ERROR
+    if (peerNodeId != existingSession->peerNodeID || peerFabricIndex != existingSession->peerFabricIndex)
+    {
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Protocols::InteractionModel::Status::ConstraintError);
```

Comment:
Does not match spec.  Also, we should consider factoring this duplicated "check for existing session that matches ctx and session id and there isn't one add status" bit into a helper function.